code/script.js

```diff
@@ -32,16 +37,177 @@ const showMessage = (message, sender) => {
 
 // Starts here
 const greeting = () => {
-  showMessage(`Hello there, What's your name?`, 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
-}
 
-// Set up your eventlisteners here
+  showMessage(`Hello Skier, what's your name?`, 'bot')
+}
 
-// When website loaded, chatbot asks first question.
-// normally we would invoke a function like this:
-// greeting()
-// But if we want to add a little delay to it, we can wrap it in a setTimeout:
-// setTimeout(functionName, timeToWaitInMilliSeconds)
 // This means the greeting function will be called one second after the website is loaded.
-setTimeout(greeting, 1000)
+setTimeout(greeting, 900)
+
+const reply = (input) => {
+  showMessage(input, 'user')
+}
+
+
+let flag = 'name'
+sendButton.addEventListener('click', () => {
+  if (flag === 'name') {
+    reply(userInput.value)
+    showMessage(`Thank you for your interest, ${userInput.value}. May I know your age?`, 'bot')
+    flag = 'age'
+  }
+  else if (flag = 'age') {
+    reply(userInput.value)
+    if (isNumeric(userInput.value) === false) {
+      showMessage(`Sorry, please only add digits.`, 'bot')
+      showMessage(` May I know your age?`, 'bot')
+    } else {
+      if (userInput.value <= 25) {
+        showMessage(`Sorry, we only offer courses for adultier adults who are older than 25 years old. :)`, 'bot')
+        setTimeout(() => location.reload(), 1000)
+        return false;
+      } else {
+        return setTimeout(skiStyle, 900)
+      }
+    }
+  }
+})
+
+function isNumeric(val) {
+  return /^-?\d+$/.test(val);
```

Comment:
Interesting, what does this do?

Reply:
Sorry I just saw this now! This is Regular Expression. I used this to check if what the user typed was number or not. I find [this website](https://bytes.com/topic/javascript/answers/150159-what-does-mean-d-test-el-value) very helpful if you want to read more about this. :)